clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -810,7 +810,7 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
   // Only enable CGProfilePass when using integrated assembler, since
   // non-integrated assemblers don't recognize .cgprofile section.
   PTO.CallGraphProfile = !CodeGenOpts.DisableIntegratedAS;
-  PTO.UnifiedLTO = CodeGenOpts.UnifiedLTO;
+  PTO.UnifiedLTO = CodeGenOpts.UnifiedLTO || CodeGenOpts.FatLTO;
```

Comment:
Ah, I remember why I did this now.  The code in `Driver/Toolchains/Clang.cpp` set things nicely from `clang`, but didn't handle `cc1`. I believe that is because these are both CodeGenOpts defined in the TableGen files.

This feels like it ought to be simple, but the driver bits have never been that intuitive to me. I *think* if I wanted to handle this directly, then I'd need to supply the implementation for both FatLTO and UnifiedLTO instead of relying on the TableGened impl, which seems worse than the current version. 

Do you have any thoughts on a nicer way to avoid the redundant code here?  Maybe we don't need the changes to `Driver/Toolchains/Clang.cpp`?

Reply:
Not sure I follow. Isn't the change in Driver/Toolchains/Clang.cpp going to ensure that -funified-lto is also passed to the cc1 invocation, and thus both options should be set in CodeGenOpts during the cc1 invocation?

Reply:
Exactly, but that doesn't change the behavior for cc1 invocations themselves ... or is that the expected behavior?

Reply:
Ah ok. I guess that would be expected. Whether it is test friendly is another question, but in general the driver does a lot of option set up for the cc1 invocation.

Reply:
Thanks for the clarification. In that case I guess I can go back to my original plan and just update the test invocations accordingly.

Reply:
Actually, I believe the cc1 parsing is handled by ParseCodeGenArgs in clang/lib/Frontend/CompilerInvocation.cpp. You can potentially add something in there to ensure UnifiedLTO is set with FatLTO? Or give your error there that UnifiedLTO must be specified with Fat LTO.

Reply:
Thank you! that was exactly the kind of place I was hoping to find.